components/raftstore-v2/src/operation/ready/snapshot.rs

```diff
@@ -132,9 +132,24 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
             self.schedule_apply_fsm(ctx);
             self.storage_mut().on_applied_snapshot();
             self.raft_group_mut().advance_apply_to(persisted_index);
+            {
+                let mut meta = ctx.store_meta.lock().unwrap();
+                meta.tablet_caches.insert(region_id, self.tablet().clone());
+                meta.readers
+                    .insert(region_id, self.generate_read_delegate());
+                meta.region_read_progress
+                    .insert(region_id, self.read_progress().clone());
+            }
             self.read_progress_mut()
                 .update_applied_core(persisted_index);
-            info!(self.logger, "apply tablet snapshot completely");
+            match self.storage_mut().split_init_mut().take() {
```

Comment:
No need to check init.scheduled?

Reply:
No, `scheduled` just mean the message is sent to peer. No matter whether it's scheduled or not, `Some` means split has happened and need to be acked.

Reply:
The assertion should be removed. It's OK the snapshot is not created by split. As long as the peer is initialized, it can report to the leader split is finished.